### PR TITLE
Allow for pluggable public dirs and service ui skins

### PR DESCRIPTION
**Pull request by Fryguy**
@simaishi Please review.

Depends on https://github.com/ManageIQ/manageiq/pull/20120

### CODE REVIEW

**Comment by Fryguy** [comment]
Note that I think from a development perspective we need some of these things as rake tasks from the individual repos, however for now we can keep it here, and then extract those later.

File: build_scripts/manageiq_tar_build.rb [hunk 1]
```diff
@@ -59,7 +65,9 @@ def precompile_sti_loader
   end
 
   def build_service_ui
-    Dir.chdir(app_dir.join("manageiq-ui-service")) do
+    symlink_plugin_paths("manageiq-ui-service", ui_service_dir)
```
Comment (Fryguy) [inline build_scripts/manageiq_tar_build.rb:68]: This, in particular should probably be a rake task as part of manageiq-ui-service itself (but future PR)

**Comment by simaishi** [comment]
This fails on the build machine...

```
Could not load database configuration. No such file - ["config/database.yml"]
```

**Comment by Fryguy** [comment]
oh it must need environment for rails runner...I'll have to write a dedicated rake task.

File: build_scripts/manageiq_tar_build.rb [hunk 2]
```diff
@@ -1,13 +1,15 @@
+require 'fileutils'
 require 'pathname'
 require 'yaml'
 
 class ManageIQTarBuild
-  attr_reader :app_dir, :cfg_dir, :miq_dir, :options
+  attr_reader :app_dir, :cfg_dir, :miq_dir, :ui_service_dir :options
```
Comment (chessbyte) [inline build_scripts/manageiq_tar_build.rb:6]: missing a comma?

**Comment by Fryguy** [comment]
Depends on https://github.com/ManageIQ/manageiq/pull/20120

File: lib/manageiq/rpm_build/generate_core.rb [hunk 3]
```diff
@@ -23,6 +26,10 @@ def release_file
         File.write(miq_dir.join("RELEASE"), RELEASE)
       end
 
+      def link_plugin_public_dirs
+        symlink_plugin_paths("public", miq_dir.join("public"))
```
Comment (Fryguy) [inline lib/manageiq/rpm_build/generate_core.rb:30]: @simaishi I noticed the following line of code, which might cause this to be an issue:

https://github.com/ManageIQ/manageiq-rpm_build/blob/a8d37edafeba9d751c53a6acb72de47a46d71bd2/lib/manageiq/rpm_build/generate_tar_files.rb#L22

Can we discuss this?  I think I understand why, but I'm thinking there might be a better way to do it there, so we don't have to enumerate things individually.

Comment (simaishi) [reply lib/manageiq/rpm_build/generate_core.rb:30]: https://github.com/ManageIQ/manageiq-rpm_build/pull/18 removes those 2 lines. Working on PR feedback, should be ready for another review soon.

**Pull request merged**